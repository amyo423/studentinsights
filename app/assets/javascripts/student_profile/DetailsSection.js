import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';


// UI component, renders box and title for section in details of profile page
export default function DetailsSection({title, className, anchorId, children}) {
  const classNameString = _.compact(["DetailsSection", className]).join(' ');
  return (
    <div className={classNameString} id={anchorId} style={styles.container}>
      <DetailsHeader title={title} />
      {children}
    </div>
  );
}
DetailsSection.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  anchorId: PropTypes.string,
};


// UI component, renders title with a link to jump back to the top of the page
export function DetailsHeader({title}) {
  return (
    <div style={styles.secHead}>
      <h4 style={styles.title}>
        {title}
      </h4>
      <span style={styles.navTop}>
        <a href="#">
          Back to top
        </a>
      </span>
    </div>
  );
}
DetailsHeader.propTypes = {
  title: PropTypes.string.isRequired
};

const styles = {
  title: {
    color: 'black',
    paddingBottom: 20,
    fontSize: 24
  },
  secHead: {
    display: 'flex',
    justifyContent: 'space-between',
    position: 'relative',
    bottom: 10
  },
  navTop: {
    textAlign: 'right',
    verticalAlign: 'text-top'
  },
  container: {
    width: '100%',
    marginTop: 50,
    marginLeft: 'auto',
    marginRight: 'auto',
    border: '1px solid #ccc',
    padding: '30px 30px 30px 30px',
    position: 'relative'
  }
};
