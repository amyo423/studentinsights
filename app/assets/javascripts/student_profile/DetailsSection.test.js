import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DetailsSection from './DetailsSection';


function testProps(props = {}) {
  return {
    title: 'foo title',
    anchorId: 'foo',
    className: 'FooClassName',
    children: <div>children are here!</div>,
    ...props
  };
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  const props = testProps();
  ReactDOM.render(<DetailsSection {...props} />, el);
});


it('snapshots', () => {
  const props = testProps();
  const tree = renderer
    .create(<DetailsSection {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});