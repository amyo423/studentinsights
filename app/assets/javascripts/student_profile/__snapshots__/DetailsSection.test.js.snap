// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  className="DetailsSection FooClassName"
  id="foo"
  style={
    Object {
      "border": "1px solid #ccc",
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": 50,
      "padding": "30px 30px 30px 30px",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "bottom": 10,
        "display": "flex",
        "justifyContent": "space-between",
        "position": "relative",
      }
    }
  >
    <h4
      style={
        Object {
          "color": "black",
          "fontSize": 24,
          "paddingBottom": 20,
        }
      }
    >
      foo title
    </h4>
    <span
      style={
        Object {
          "textAlign": "right",
          "verticalAlign": "text-top",
        }
      }
    >
      <a
        href="#"
      >
        Back to top
      </a>
    </span>
  </div>
  <div>
    children are here!
  </div>
</div>
`;
